src/python/pants/core/goals/run.py

```diff
@@ -65,10 +73,15 @@ def __init__(
         self.extra_env = FrozenDict(extra_env or {})
 
 
+@dataclass(frozen=True)
+class RunConsoleScriptRequest:
+    console_script: str
+
+
 class RunSubsystem(GoalSubsystem):
```

Comment:
> Ah, right. Yes, it does work using the 3rdparty dep and pex_binary route. The awkwardness is simply that if I already have it listed in plugins, it is there ready to be used. Without this, I'd have to add some targets to a BUILD file to get there. This is boilerplate required per script I want to call. As a plugin author, it would be a hassle to ask users to add this to their BUILD files in order to access the added tooling.
> 
> Granted, the overhead isn't much, but it adds BUILD scaffolding noise that is avoidable, as well as keeping the configuration DRY, as otherwise, any version constraints will be duplicated (not in conflict though, you could use two different versions).
> 
> So, this is more of a nice feature, than required to get where I want..

And with synthetic targets, I think this point is pretty much moot by now, as the plugin might register its own set of "provided" `pex_binary` targets that can be used right away.

Reply:
Now, I wonder why that would not just be another `goal`...